routers/repo/issue.go

```diff
@@ -1237,6 +1237,67 @@ func UpdateIssueAssignee(ctx *context.Context) {
 	})
 }
 
+// updatePullReviewRequest change pull's request reviewers
+func updatePullReviewRequest(ctx *context.Context, isReRequest bool) {
+	issues := getActionIssues(ctx)
+	if ctx.Written() {
+		return
+	}
+
+	reviewID := ctx.QueryInt64("id")
+	action := ctx.Query("action")
+
+	if action != "" {
+		ctx.ServerError("updatePullReviewRequest", fmt.Errorf("action should not be \"%s\"", action))
+		return
+	}
+
+	for _, issue := range issues {
+		if issue.IsPull {
+
+			if isReRequest && issue.PosterID != ctx.User.ID {
```

Comment:
Hello, The permissions style which has been writen now is same as github, And I think it is reasonable. Because in github's seting , It's a offical manager work to Assign reviewer for a pr , which is same as Assign , Lable and Milestone.
so it's reasonable to limit to write permission. and add a re-request fun which is only for author to call after a review is exit. That's just my thinking , It should have more disscussion. or may be way can provide two mode for "owner" of a
repo to choose.

Reply:
Hello, I have maken my thought as an issue:#10799

Reply:
OK, updating my list, I still think the reviewers should be able to dismiss themselves, regardless of what permissions they have.

Also, people with write permissions should not be able to dismiss any review (I as a maintainer am not allowed to dismiss other maintainer's review here in Github). Admins/owners should.

Having said that, Gitea does have its own personality, so some tasks are not entirely resolved the same way as Github. For example, reviewer whitelists are handled separately from PR write access, so we must be careful on what we can copy and what we should reconsider.